.cursor/.rules/planning-overview.mdc

```diff
@@ -0,0 +1,30 @@
+---
+description: High-level priorities and decision heuristics for Cursor Plan Mode
+globs:
+alwaysApply: true
+---
+
+Principles
+
+- Prefer minimal, focused edits that align with existing patterns.
+- Optimize for readability and maintainability over cleverness.
+- Encode domain decisions in small utilities/hooks to avoid repetition.
+- Prefer composition over inheritance; avoid deep abstractions without need.
+
+Decision Heuristics
+
+- State: Use React Context for cross-cutting app/data state; use Zustand only for ephemeral UI state.
+- Data fetching: Use Apollo Client; default to cache-first; use `onCompleted`/`onError` for `useQuery` side-effects and async/await with try/catch for `useMutation`/`useLazyQuery`.
```

Comment:
Apollo is deprecating the lifecycle hooks for `useQuery` and `useLazyQuery`, might be worth removing that part. https://github.com/apollographql/apollo-client/issues/12352